src/xpcc/architecture/interface/block_device.hpp

```diff
@@ -0,0 +1,104 @@
+// coding: utf-8
+/* Copyright (c) 2017, Raphael Lehmann
+ * All Rights Reserved.
+ *
+ * The file is part of the xpcc library and is released under the 3-clause BSD
+ * license. See the file `LICENSE` for the full license governing this code.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef XPCC_INTERFACE_BLOCK_DEVICE_HPP
+#define XPCC_INTERFACE_BLOCK_DEVICE_HPP
+
+#include <stdint.h>
+#include <xpcc/architecture/interface.hpp>
+#include <xpcc/processing/resumable.hpp>
+
+/**
+ * @ingroup		interface
+ * @defgroup	storage	Data storage
+ */
+
+namespace xpcc
+{
+
+/**
+ * Interface of a Block Device
+ *
+ * Access to storage devices like flash chips
+ *
+ * @author	Raphael Lehmann
+ * @ingroup	storage
+ */
+#ifdef __DOXYGEN__
+class BlockDevice : protected xpcc::NestedResumable<3>
```

Comment:
Don't inherit the interface from `*Resumable`, since that is implementation detail.